org.adempiere.base/src/org/compiere/model/MProcessDrillRule.java

```diff
@@ -211,4 +221,40 @@ public PO markImmutable() {
 		makeImmutable();
 		return this;
 	}
+	
+	/**
+	 * Are all mandatory parameters defined among the Drill Rule Parameters
+	 * @param processID
+	 * @return true - all mandatory parameters are set; false - at least one mandatory parameter is not set
+	 */
+	public boolean allMandatoryParaSet() {
+		boolean isValid = false;
+		MProcess process = new MProcess(Env.getCtx(), getAD_Process_ID(), null);
+		for(MProcessPara processPara : process.getParameters()) {
+			if(processPara.isMandatory()) {
+				for(MProcessDrillRulePara drillRulePara : getParameters(true)) {
+					if(drillRulePara.getAD_Process_Para_ID() == processPara.getAD_Process_Para_ID()) {
+						String defPara = drillRulePara.getParameterDefault();
+						String defParaTo = drillRulePara.getParameterToDefault();
+						isValid = (processPara.isRange() && (!Util.isEmpty(defPara)) || (!Util.isEmpty(defParaTo))) ||
+								(!processPara.isRange() && (!Util.isEmpty(defPara)));
+						break;
+					}
+				}
+				if(!isValid)
+					return false;
+				isValid = false;
+			}
+		}
+		return true;
+	}
+	
+	/**
+	 * Validate Drill Rule - update IsValid
+	 */
+	public void validate() {
+		String sql = " UPDATE " + MProcessDrillRule.Table_Name + " SET " + MProcessDrillRule.COLUMNNAME_IsValid + " = ? "
+				+ " WHERE " + MProcessDrillRule.Table_Name + "_ID = ?";
+		DB.executeUpdateEx(sql, new Object[] {allMandatoryParaSet() ? "Y" : "N", getAD_Process_DrillRule_ID()}, get_TrxName());
```

Comment:
It seems strange to use afterSave and executeUpdateEx here. Should be much simpler to just do it in beforeSave and set IsValid to true of false.

Reply:
ok, I've rewritten and moved it to beforeSave